packages/dev/mcp/src/common/parser.ts

```diff
@@ -0,0 +1,52 @@
+import type {SectionInfo} from './types.js';
+
+export function parseSectionsFromMarkdown(lines: string[]): SectionInfo[] {
```

Comment:
we have remark, so you could do something like

```
    const tree = unified().use(remarkParse).parse(lines.join('\n'));

    const headings = [];

    visit(tree, 'heading', (node) => {
      const headingText = node.children[0].value; // Assuming simple text headings
      const level = node.depth;
      headings.push({ text: headingText, level: level });
    });
```

based on usage https://github.com/adobe/react-spectrum/blob/970f8a3965d198fbb9d0762c250d3857bd63f1cf/packages/dev/s2-docs/scripts/generateMarkdownDocs.mjs#L10

then you could use level to determine if it's the title/name

Reply:
Same, can improve these in a follow-up, just wanted to include the package refactor in this PR.